src/App.tsx

```diff
@@ -1,10 +1,90 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { useTodo } from './hooks/useTodo';
 
-const USER_ID = 0;
+const USER_ID = 11340;
 
 export const App: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    isChecked,
+    setIsChecked,
+  } = useTodo();
+
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      });
+  }, []);
+
+  setIsChecked(todos.every(todo => todo.completed) && todos.length > 0);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newTitle = newTodoTitle.trim();
+
+    if (!newTitle) {
+      setNewTodoTitle('');
+
+      return;
+    }
+
+    const newTodo = {
+      id: +new Date(),
+      userId: USER_ID,
+      title: (newTitle),
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
+    setNewTodoTitle('');
+  };
+
+  const handleCheckAllTodos = () => {
+    setTodos(todos.map(todo => ({
```

Comment:
you can use previous values of your state
```suggestion
    setTodos((prev) => prev.map(todo => ({
```